package-scripts/datadog-agent/postinst

```diff
@@ -57,10 +57,11 @@ if which chkconfig >/dev/null 2>&1; then
   chkconfig --add datadog-agent
 fi
 
+MISSING_CONF_MSG="\033[34m\n* No /etc/dd-agent/datadog.conf file has been found.\033[0m"
 if which invoke-rc.d >/dev/null 2>&1; then
-    invoke-rc.d datadog-agent restart
+    invoke-rc.d datadog-agent restart || echo "$MISSING_CONF_MSG"
 else
-    /etc/init.d/datadog-agent restart
+    /etc/init.d/datadog-agent restart || echo "$MISSING_CONF_MSG"
```

Comment:
When using the `||` operator, you swallow all non-zero exit codes. In the agent init script, the "missing config" case is something that is handled differently see https://github.com/DataDog/dd-agent/blob/master/packaging/debian/datadog-agent.init#L71-L75
In case the restart failed because of some other reason and the init script exits with a non-zero and `!=3` error code, you'll still output the `MISSING_CONF_MSG` and exit gracefully ; I think that's not what's expected.